dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/communityManager/QueryDataRequestsTest.kt

```diff
@@ -88,163 +92,343 @@ class QueryDataRequestsTest {
-            assertTrue(dataRequests.all { it.userEmailAddress == null })
+            val dataRequests = api.getDataRequests(datalandCompanyId = companyIdA)
+            assertTrue(dataRequests.all { it.userEmailAddress != null })
         }
 
         val queryingUser = TechnicalUser.Reader
 
         withTechnicalUser(queryingUser) {
-            assertAccessDeniedWrapper { requestControllerApi.getDataRequests(datalandCompanyId = companyIdA) }
+            assertAccessDeniedWrapper { api.getDataRequests(datalandCompanyId = companyIdA) }
         }
 
         assignCompanyOwnershipToUser(companyIdA, queryingUser.technicalUserId)
 
         withTechnicalUser(queryingUser) {
-            val dataRequests = requestControllerApi.getDataRequests(datalandCompanyId = companyIdA)
+            val dataRequests = api.getDataRequests(datalandCompanyId = companyIdA)
             assertEquals(2, dataRequests.size)
             assertTrue(dataRequests.all { it.userEmailAddress != null })
         }
 
         removeCompanyOwnershipFromUser(companyIdA, queryingUser.technicalUserId)
 
         withTechnicalUser(queryingUser) {
-            assertAccessDeniedWrapper { requestControllerApi.getDataRequests(datalandCompanyId = companyIdA) }
+            assertAccessDeniedWrapper { api.getDataRequests(datalandCompanyId = companyIdA) }
+        }
+    }
+
+    @Test
+    fun `query data requests and assert that email address is not visible for public data requests`() {
+        val queryingUser = TechnicalUser.Reader
+
+        withTechnicalUser(queryingUser) {
+            assertAccessDeniedWrapper { api.getDataRequests(datalandCompanyId = companyIdB) }
         }
+
+        assignCompanyOwnershipToUser(companyIdB, queryingUser.technicalUserId)
+
+        withTechnicalUser(queryingUser) {
+            val dataRequests = api.getDataRequests(datalandCompanyId = companyIdB)
+            assertEquals(1, dataRequests.size)
+            assertTrue(dataRequests.all { it.userEmailAddress == null })
+        }
+
+        removeCompanyOwnershipFromUser(companyIdB, queryingUser.technicalUserId)
+
+        withTechnicalUser(queryingUser) {
+            assertAccessDeniedWrapper { api.getDataRequests(datalandCompanyId = companyIdB) }
+        }
+    }
+
+    @Test
+    fun `query data requests with combined filter and assert that the expected results are being retrieved`() {
+        val combinedQueryResults =
+            api.getDataRequests(
+                dataType = listOf(sfdrType, p2pType, vsmeType),
+                requestStatus = setOf(RequestStatus.Open, RequestStatus.Resolved),
+                accessStatus = setOf(AccessStatus.Pending),
+                reportingPeriod = "2023",
+                chunkSize = chunkSize,
+            )
+                .filter { it.creationTimestamp > timestampBeforePost }
+        assertEquals(1, combinedQueryResults.size)
+    }
+
+    private val sfdrType2 = RequestControllerApi.DataTypeGetNumberOfRequests.sfdr
```

Comment:
Why are they called "sfdrType2"? What does the 2 mean?

Reply:
The dataType Enum is reexported (from the backend OpenAPI) and every time this happens the type gets another name. Even if the enum is equal, it is not the same for Kotlin `sfdrType != sfdrType2`. (There is a subtask to investigate this further, but this is actually a known issue.)

Reply:
Ok, but i the naming with ...2 is not optimal. They should be called something else, because reading the variable it is not clear what this means.

Reply:
Was hälst du von den Namen sfdrRequestNumber für die Variablen?

Reply:
Bzw. eher sfdrRequestNumberDataType oder ähnlich

Reply:
Should be changed.